costmap_2d/plugins/obstacle_layer.cpp

```diff
@@ -525,6 +531,31 @@ void ObstacleLayer::raytraceFreespace(const Observation& clearing_observation, d
   {
     double wx = cloud.points[i].x;
     double wy = cloud.points[i].y;
+    if(clearing_observation.start_clearing_range_ > 0)
+    {
+      // If this sensor start seeing obstacle at a certain range, we must move x0,y0 at this
+      // range from the sensor (following the line going from the sensor to the point)
+      double obstacle_range = hypot(wx-ox, wy-oy);
+      if(obstacle_range < clearing_observation.start_clearing_range_)
+      {
+        // This may happens if parameter start_clearing_range was set to a range lesser than the minimal sensor range.
+        // In this particular case, we just don't set free spaces (after displaying a warning).
+        ROS_WARN_THROTTLE(
+                1.0, "Your start_clearing_range parameter seems to be too short because an observation was returned by"
+                "your sensor in this range. Cells between sensor and this observation was not cleared");
```

Comment:
I'm not a fan of this log message -- there are certainly sensors that report 0 when they cannot see something, and this would spam the crap out of the console/logs. I would suggest we either remove this, or make it a ROS_DEBUG_NAMED_THROTTLE and increase the throttle to something like 60.0.